libs/s25main/desktops/dskOptions.cpp

```diff
@@ -663,3 +689,89 @@ void dskOptions::loadVideoModes()
     // Sort by aspect ratio
     std::sort(video_modes.begin(), video_modes.end(), cmpVideoModes);
 }
+
+void dskOptions::Msg_ScreenResize(const ScreenResizeEvent& sr)
+{
+    Desktop::Msg_ScreenResize(sr);
+    updateGuiScale();
+}
+
+bool dskOptions::Msg_WheelUp(const MouseCoords& mc)
+{
+    if(VIDEODRIVER.GetModKeyState().ctrl)
+    {
+        scrollGuiScale(true);
+        return true;
+    } else
+        return Desktop::Msg_WheelUp(mc);
+}
+
+bool dskOptions::Msg_WheelDown(const MouseCoords& mc)
+{
+    if(VIDEODRIVER.GetModKeyState().ctrl)
+    {
+        scrollGuiScale(false);
+        return true;
+    } else
+        return Desktop::Msg_WheelDown(mc);
+}
+
+void dskOptions::updateGuiScale()
+{
+    // generate GUI scale percentages in 10% increments
+    constexpr auto stepSize = 10u;
+    const auto roundGuiScale = [=](unsigned percent) {
+        return helpers::iround<unsigned>(static_cast<float>(percent) / stepSize) * stepSize;
+    };
+
+    unsigned minPercent, maxPercent, recommendedPercent;
+    VIDEODRIVER.getGuiScaleRange(&minPercent, &maxPercent, &recommendedPercent);
+    const auto recommendedPercentRounded = roundGuiScale(recommendedPercent);
+    auto* group = GetCtrl<ctrlGroup>(ID_grpGraphics);
+    auto* combo = group->GetCtrl<ctrlComboBox>(ID_cbGuiScale);
```

Comment:
Any reason not to use the following as `group` is only used in this single line:

```suggestion
    auto* combo = GetCtrl<ctrlGroup>(ID_grpGraphics)->GetCtrl<ctrlComboBox>(ID_cbGuiScale);
```